cdm/misc/src/main/java/ucar/nc2/geotiff/GeotiffWriter.java

```diff
@@ -261,6 +271,91 @@ private void writeMetadata(boolean greyScale, double xStart, double yStart, doub
     geotiff.writeMetadata(imageNumber);
   }
 
+  /**
+   * Get a copy of the current colormap as a 1-D 3*256 element array (or null).
+   *
+   * All 256 red values first, then green, then blue.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 65535 is maximum intensity (due to geotiff conventions).
+   * This function is intended for debugging and testing.
+   */
+  public int[] getColorTable() {
+    if (colorTable == null) {
+      return null;
+    } else {
+      return colorTable.clone();
+    }
+  }
+
+  /**
+   * Provide a colormap in the form of a mapping of the pixel value to the rgb triplet.
+   * Assumes an RGB of {255, 255, 255} for any values not specified.
+   *
+   * Pass null to unset the colorTable.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 255 is maximum intensity. Values outside that range will
+   * be floored/ceilinged to the [0, 255] range. The color table is also assumed to be for pixel values
+   * between 0 and 255.
+   */
+  public void setColorTable(Map<Integer, Color> colorMap) {
+    setColorTable(colorMap, new Color(255, 255, 255));
+  }
+
+  /**
+   * Provide a colormap in the form of a mapping of the pixel value to the rgb triplet.
+   * Provide a default RGB triplet for any values not specified.
+   *
+   * Pass null to unset the colorTable.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 255 is maximum intensity. Values outside that range will
+   * be floored/ceilinged to the [0, 255] range. The color table is also assumed to be for pixel values
+   * between 0 and 255.
+   */
+  public void setColorTable(Map<Integer, Color> colorMap, Color defaultRGB) {
+    if (colorMap == null) {
+        colorTable = null;
+        return;
+    }
+
+    // FIXME: This isn't quite right because this assumes that the data being written is
+    // unsigned bytes, but the tiff spec says that it should be sized to the width of
+    // the data type, but I would need to know the data type, which this writer doesn't know.
```

Comment:
Might also want to consider optional support for the alpha channel? I haven't tried making a paletted geotiff with a fourth channel yet.

Reply:
Thinking some more on this FIXME of mine. The only place where one knows what the dtype is going to be is in `writeGrid()` (and consequently, `writeMetadata()`). The GeotiffWriter class acts more like a state-less methods class rather than some sort of stateful "builder" class. Maybe my `setColorTable()` approach is an anti-pattern here and I really should be making a `ColorTable` class that can get passed into `writeGrid()` and `writeMetadata()`? Then, when the time comes to construct the colortable array, I'll know the width of the output dtype and can construct it appropriately?